components/service-mgt/module-mgt/org.wso2.carbon.module.mgt/src/main/java/org/wso2/carbon/module/mgt/service/ModuleAdminService.java

```diff
@@ -73,15 +79,21 @@
     private static final String RAHAS_MODULE_NAME = "rahas";
     
     private PersistenceFactory pf;
+    
+    private Registry registry;
 
     public ModuleAdminService() throws Exception {
         this.axisConfig = getAxisConfig();
         pf = PersistenceFactory.getInstance(axisConfig);
+        registry = DataHolder.getRegistryService().getConfigSystemRegistry(
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId());
     }
 
     public ModuleAdminService(AxisConfiguration ac) throws Exception {
         this.axisConfig = ac;
         pf = PersistenceFactory.getInstance(axisConfig);
+        registry = DataHolder.getRegistryService().getConfigSystemRegistry(
```

Comment:
Assigning a service object to a class variable is not a good practice. Instead get it from the data holder and do a null check before using it.